community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/TransactionHeaders.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * The creation of the byte array is encapsulated in the {@link TransactionHeadersArrayBuilder}. The parser of the array
+ * is encapsulated in this class. The general usage pattern is that when writing out the header one uses the Builder
+ * to set it up as required and then pass the resulting array to the {@link TransactionAppender} via a
+ * {@link TransactionRepresentation}. Reading back the information requires getting the header array from the
+ * {@link TransactionRepresentation} and feeding it into an instance of this class to get the encoded information.
+ *
+ * The version information kept in the first byte is for future proofing against format changes. Right now there is
+ * only one version, the current one, so it is not used anywhere and defaults always to
+ * TransactionHeaders.CURRENT_VERSION
+ */
+public class TransactionHeaders
+{
+    public static final byte CURRENT_VERSION = 0;
+    private static final byte[] DEFAULT_FOR_NON_EXISTING_ID = new byte[0];
+
+    private final byte version;
+    private final Map<Byte, byte[]> entries = new HashMap<>();
+
+    public TransactionHeaders( byte[] bytes )
+    {
+        this.version = extractVersion( bytes );
+        extractEntries( bytes );
+    }
+
+    /**
+     * @return The number of headers discovered and parsed
+     */
+    public int size()
+    {
+        return entries.size();
+    }
+
+    /**
+     * @return The version this array has been encoded with.
+     */
+    public byte version()
+    {
+        return version;
+    }
+
+    /**
+     * @return The byte array belonging to the specified identifier
+     */
+    public byte[] forIdentifier( byte identifier )
+    {
+        return entries.containsKey( identifier ) ?
+                entries.get( identifier ) :
+                DEFAULT_FOR_NON_EXISTING_ID;
+    }
+
+    private byte extractVersion( byte[] bytes )
+    {
+        return bytes.length > 0 ? bytes[0] : CURRENT_VERSION;
+    }
+
+    private void extractEntries( byte[] bytes )
+    {
+        if ( bytes.length < 2 ) // either empty array or just with version
+        {
+            return;
+        }
+        int offset = 1; // skip position 0, it's the version
+        byte numberOfEntries = bytes[offset];
+        offset += 1;
+        for ( byte i = 0; i < numberOfEntries; i++ )
+        {
+            byte currentIdentifier = bytes[offset++];
+            int currentLength = decodeLengthFromByteArray( bytes, offset );
+            offset += Integer.BYTES; // that's how much the previous method reads
+            byte[] content = Arrays.copyOfRange( bytes, offset, offset + currentLength );
+            entries.put( currentIdentifier, content );
+
+            offset += currentLength;
+        }
+        assert offset == bytes.length; // we should have exhausted the array
+    }
+
+    private static void encodeLengthIntoByteArray( int length, byte[] bytes, int offset )
+    {
+        for ( int i = Integer.BYTES - 1; i > 0; i-- )
```

Comment:
It looks like you're really trying to avoid existing utility methods for int <-> byte[]. Would you mind keeping the byte[] as-is, but put/get data to/from it using a ByteBuffer allocated using `ByteBuffer.wrap` on that array as to reduce code volume and in extension maintenance? Perhaps it would reduce amount of code elsewhere in this class as well?